### DISCUSSION TITLE
Tracking river flow

### DISCUSSION THREAD

**Discussion by mariemccrary** [Parcels-code/Parcels#1038] [Q&A]
Hi everyone.

I'd like to use Parcels to track river volume fluxes into the Arctic Ocean. Is there a way to scale the number of particles released over the course of a run to correspond with changing riverine volume fluxes (ie large in the summer and small in the winter)?

Thanks for any insight.

Marie

**Comment by erikvansebille** [comment]
Hi Marie, thanks for your question. I think that @cpierard, @mikaelk and/or @VictorOnink have set up simulations with a particle release based on monthly river plastic fluxes in the past; they may be able to help?

**Reply by VictorOnink** [reply]
Hi Marie,
I have done a number of simulations where I scale my input according to input estimates, but I always fixed this scaling at the beginning of the simulations and this then didn't change over the course of a run. So based on my understanding of the question that isn't what you are trying to do.
On the top of my head (so not something I've actually tried), what could maybe work is that when you create your `particleset` and set the initial particle lon/lat positions, you also specify the initial time of the particle depending on at what point in the year you want to release them. So if you want to release 20 particles at some river mouth where 10 should be released for one specific month, you could for example for these 10 initialise the time value to be in August so that this is when this many particles are released (see this tutorial https://nbviewer.jupyter.org/github/OceanParcels/parcels/blob/master/parcels/examples/tutorial_delaystart.ipynb). I haven't ever implemented this, but I think it should probably work?

**Reply by mikaelk** [reply]
Yes you could do it like Victor proposed, create 3 arrays, e.g. particles_lon, particles_lat, particles_time. Then in your script, make a loop through time (e.g. days), and for each time, look how many particles are released at each river mouth. Append the lon/lat/time of each particle to the corresponding particles_lon/particles_lat/particles_time arrays. Then just like in the tutorial run
pset = ParticleSet(fieldset=fieldset, pclass=JITParticle, lon=particles_lon, lat=particles_lat, time=particles_time)
